### PR TITLE
Assume no terminal if the display width/height is 0

### PR DESCRIPTION
**Pull request by leezen**
I'm not super sure about the warning going to stderr, but that seems like the right thing. Otherwise, the user is left wondering why they aren't getting the normal interactive terminal experience when the terminal is not reporting width/height correctly.

Fixes https://github.com/pulumi/pulumi/issues/5957

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ CHANGELOG
 - [sdk/dotnet] Add collection initializers for smooth support of Union<T, U> as element type
   [#5938](https://github.com/pulumi/pulumi/pull/5938)
 
+- Fix a bug in the Go SDK that could result in dropped resource dependencies.
```
Comment (pgavlin) [inline CHANGELOG.md:9]: Was this intentional?

Comment (leezen) [reply CHANGELOG.md:9]: No... I must have done something odd when merging from master. Let me fix that up.

**Comment by lukehoban** [comment]
Does this also fix https://github.com/pulumi/pulumi/issues/5643??

**Comment by leezen** [comment]
@lukehoban I haven't had a chance to try to repro #5643. Only #4904, which I wasn't able to repro. I'll try to find some time today to repro #5643

**Review by pgavlin — APPROVED** [review]

**Pull request merged**